android-components/components/support/remotesettings/src/main/java/mozilla/components/support/remotesettings/RemoteSettingsClient.kt

```diff
@@ -0,0 +1,136 @@
+import mozilla.appservices.remotesettings.RsJsonObject
+import mozilla.components.support.base.log.logger.Logger
+import mozilla.components.support.ktx.util.writeString
+import org.json.JSONArray
+import org.json.JSONObject
+import java.io.File
+import java.io.FileReader
+
+/**
+ * Helper class to download collections from remote settings in app services.
+ *
+ * @property storageRootDirectory The top level app-local storage directory.
+ * @property serverUrl The optional url for the settings server. If not specified, the standard server will be used.
+ * @property bucketName The optional name of the bucket containing the collection on the server.
+ * If not specified, the name of the bucket "main" [or the const if updated] will be used.
+ * @property collectionName The name of the collection for the settings server.
+ */
+class RemoteSettingsClient(
+    private val storageRootDirectory: File,
+    private val serverUrl: String = "https://firefox.settings.services.mozilla.com",
+    private val bucketName: String = "main",
+    private val collectionName: String = "",
+) {
+
+    private val config = RemoteSettingsConfig(
+        serverUrl = serverUrl,
+        bucketName = bucketName,
+        collectionName = collectionName,
+    )
+
+    /**
+     * Fetches all records for the collection from remote servers and updates the local storage.
+     */
+    suspend fun fetchRemoteServerRecords(): List<RemoteSettingsRecord> = withContext(Dispatchers.IO) {
+        try {
+            val serverRecords = RemoteSettings(config).use {
+                it.getRecords().records
+            }
+            return@withContext serverRecords
+        } catch (e: RemoteSettingsException.ResponseException) {
+            emptyList()
+        } catch (e: RemoteSettingsException) {
+            Logger.error(e.message.toString())
+            emptyList()
+
+        }
+    }
+
+    private fun writeRemoteRecordsToStorage(file: File, records: List<RemoteSettingsRecord>) {
+        try {
+            val jsonArray = JSONArray()
+            records.forEach {
+                val recordJson = JSONObject().apply {
+                    put("id", it.id)
+                    put("lastModified", it.lastModified.toString())
+                    put("deleted", it.deleted)
+                    put("attachment", it.attachment)
+                    put("fields", it.fields)
+                }
+                jsonArray.put(recordJson)
+            }
+
+            val jsonObject = JSONObject()
+            jsonObject.put("records", jsonArray)
+            AtomicFile(file).writeString { jsonObject.toString() }
+        } catch (e: Exception) {
+            Logger.error(e.message.toString())
+        }
+    }
+
+    /**
+     * Reads all records for a collection that can be found in the local storage.
+     **/
+    suspend fun readStorage(): List<RemoteSettingsRecord> = withContext(Dispatchers.IO) {
+        val folder = File(storageRootDirectory, bucketName)
```

Comment:
Yes if we are offering the consumers to configure the path/name of the file, it could be hard to distinguish. Had we not done that, the same server and bucket names would still have contributed to different path , because the file name(collection name)  would have been different.

Reply:
I think we should still use the server name as part of the directory path - otherwise 2 servers with the same bucket and collection names would overlap. I don't expect to actually run into that, but it seems worth updating this line to be something like:
```
val folder = File(storageRootDirectory, serverName.stripProtocol(), bucketName)
```